### ISSUE TITLE
New match_symbol_usage argument to undesirable_function_linter

### ISSUE THREAD

**Issue by MichaelChirico** [jimhester/lintr#814]
We have a function `foo()` that we'd like to lint. However, `foo()` comes from the package `foo`, which is still OK, so we have the following unfortunate situation:

```
library(foo) # triggers a lint, but shouldn't
foo::foo()  # triggers a lint, as it should
```

The issue is that `undesirable_function_linter` by default also looks for `SYMBOL` usage of input functions, which is good for catching usage within `sapply`, `purrr::map`, etc.

So we can add a new argument to toggle whether `SYMBOL` is included, and run linting with

```
with_defaults(
  undesirable_function_linter(c("return")), # match_symbol_usage = TRUE by default
  undesirable_function_linter(c("foo"), match_symbol_usage = FALSE)
)
```

Happy to file the PR if that sounds like a good idea.

**Comment by MichaelChirico** [comment]
alternatively we could treat this as a bug since it basically is a false positive. solution would be to switch to XML search in the linter & exclude parent library/require calls...

**Comment by kpagacz** [comment]
I personally would treat it as a false positive.

Do you mind if I started working on this?

**Comment by MichaelChirico** [comment]
sure! PR is a good place to continue the discussion
